server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java

```diff
@@ -467,10 +488,27 @@ public void uploadMetadata(
             );
             try {
                 IndexOutput indexOutput = storeDirectory.createOutput(metadataFilename, IOContext.DEFAULT);
+                List<SegmentCommitInfo> infos = segmentInfosSnapshot.asList();
+                Map<String, Version> segmentToLuceneVersion = new HashMap<>();
+                for (SegmentCommitInfo segmentCommitInfo : infos) {
```

Comment:
I understand why we are making this change keeping it forward looking but as long as we don't support enabling remote store on existing index, we can always rely on getting current Lucene version that we are using to upload new files. For existing files in the remote store, metadata will anyway have this info.

Reply:
We would need to depend on getting the Lucene version from `SegmentInfos` for snapshot interoperability

Reply:
Hey Sachin,
Resolving this. Please reopen if any changes are needed here.
Thanks!